servicetest/filesystem/test_filesystem.py

```diff
@@ -251,15 +251,94 @@ def test_tmpfs_writebuffer_size_enabled(self):
 
     @pytest.mark.xfail("platform.release() <= \"3.18.0\"")
-    def test_tmpfs_writebuffer_size_disabled(self):
+    def test_tmpfs_writebuffer_size_enabled_10mb(self):
         """ The SoftwareContainer will create a tmpfs in the location of the
             containers temporary directory location if the write buffer is
             enabled. Test that the tmpfs mounted in the containers temporary
             directory structure works as expected.
 
-            Note: The filesystem is hard coded to 100MB at the time this test
-            was written, but will be configurable via the DATA field in a
-            short while
+            This test will create a 10 MB tmpfs which will be tested that it
+            behaves as expected.
+        """
+
+        absoluteTestFile = os.path.join("/tmp/container/SC-0/rootfs-upper",
+                                        TESTFILE)
+
+        ca = Container()
+
+        DATA[Container.CONFIG] = '[{"enableWriteBuffer": true, \
+                                 "enableTemporaryFileSystemWriteBuffer": true, \
+                                 "temporaryFileSystemSize": 10485760}]'
+
+        try:
+            ca.start(DATA)
+
+            partitions = psutil.disk_partitions(all=True)
+            interesting_partition = False
+            for part in partitions:
+                if part.mountpoint == '/tmp/container/SC-0':
+                    interesting_partition = part
+                    break
+            assert interesting_partition is not False
+
+            with open(absoluteTestFile, "w") as f:
+                f.write("w" * 9 * 1024 * 1024)
+
+            assert os.path.getsize(absoluteTestFile) >= 8 * 1024 * 1024
+            assert os.path.getsize(absoluteTestFile) <= 10 * 1024 * 1024
+
+            with open(absoluteTestFile, "w") as f:
+                with pytest.raises(IOError):
+                    f.write("w" * 105 * 1024 * 1024)
+        finally:
+            ca.terminate()
+
+        if os.path.exists(absoluteTestFile):
+            os.remove(absoluteTestFile)
+
+    @pytest.mark.xfail("platform.release() <= \"3.18.0\"")
+    def test_tmpfs_writebuffer_size_bad_config(self):
+        """ This test will test various bad configuration combinations and
+            make sure that SoftwareContainer-Agent behaves as expected when
+            the bad configuration is sent to it.
+        """
+
+        absoluteTestFile = os.path.join("/tmp/container/SC-0/rootfs-upper",
+                                        TESTFILE)
+
+        ca = Container()
+
+        # This run should raise a DBus exception as we don't have a
+        # temporaryFileSystemSize defined
+        DATA[Container.CONFIG] = '[{"enableWriteBuffer": true, \
+                                 "enableTemporaryFileSystemWriteBuffer": true}]'
+
+        try:
+            ca.start(DATA)
+        finally:
+            ca.terminate()
+
+        # This run should work, but no temporary filesystem will be setup as
```

Comment:
No good reason to be honest except trying to shorten down runtime. I'll break it out.